### PR TITLE
Add support for responsive image

### PR DESCRIPTION
**Pull request by ghost**
Responsive image using cloudinary-core functions, like it's done in the Angular SDK. 
Wanted to implement this from scratch before refactoring this feature in the React SDK.
Added very minor test, to be expanded in the future, and a working example.

### CODE REVIEW

File: src/Image.svelte [hunk 1]
```diff
@@ -1,10 +1,23 @@
 <script>
-  /**
-   * Cloudinary Image Component
-   */
+    /**
+     * Cloudinary Image Component
+     */
+    import {onMount, afterUpdate} from 'svelte';
```
Comment (asisayag2) [inline src/Image.svelte]: What is `onMount` used for?

File: src/Image.svelte [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    import {getImageTag, responsive} from "./utils";
```
Comment (asisayag2) [inline src/Image.svelte]: Can we rename `responsive` to something more explanatory? like `makeTagResponsive`?

File: src/Image.svelte [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+    const bindImage = (node) => (imgElement = node);
 
-  import { getImageTag } from "./utils";
-  $: imageTag = getImageTag($$props);
+    let imgElement; // Reference to underlying <img> element
+    let isResponsive;
```
Comment (asisayag2) [inline src/Image.svelte]: Why not declaring inside the method?

File: src/Image.svelte [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+
+    $: attributes = getImageTag($$props).attributes();
```
Comment (asisayag2) [inline src/Image.svelte]: Something is not very clear to me... you get the Image tag, take its attributes, but at the end the attributes of the img below is based on the same `attributes`. I probably miss something

File: src/Image.svelte [hunk 5]
```diff
@@ -1,10 +1,23 @@
...
+
+    afterUpdate(() => {
+        // Run responsive() only when props.responsive is updated to truthy value
+        isResponsive = $$props.responsive && imgElement && !isResponsive;
```
Comment (asisayag2) [inline src/Image.svelte]: Can you explain the `!isResponsive` at the end?

Comment (ghost) [reply src/Image.svelte]: Removed this

File: src/Image.svelte (hunk 4 above)
Comment (ghost) [reply src/Image.svelte]: Added comments to the code

File: src/Image.svelte (hunk 3 above)
Comment (ghost) [reply src/Image.svelte]: See added comments in the code

File: src/Image.svelte (hunk 2 above)
Comment (ghost) [reply src/Image.svelte]: renamed to makeElementResponsive

File: src/Image.svelte (hunk 1 above)
Comment (ghost) [reply src/Image.svelte]: removed

**Review by asisayag2 — APPROVED** [review]

**Review by strausr — APPROVED** [review]

File: examples/public/global.css [hunk 6]
```diff
@@ -64,3 +64,8 @@ button:not(:disabled):active {
 button:focus {
 	border-color: #666;
 }
+
+.cld-responsive{
```
Comment (strausr) [inline examples/public/global.css:68]: I see that code-wrapper i used for the testing but what are the other styling used for?

Comment (ghost) [reply examples/public/global.css:68]: global.css has default styles that come with the svelte app template, i added .cld-responsive

**Pull request merged**